rust/toolchain.bzl

```diff
@@ -71,26 +71,88 @@ def _make_libstd_and_allocator_ccinfo(ctx, rust_lib, allocator_library):
                 print("File partitioned: {}".format(f.basename))
             fail("rust_toolchain couldn't properly partition rlibs in rust_lib. Partitioned {} out of {} files. This is probably a bug in the rule implementation.".format(partitioned_files_len, len(dot_a_files)))
 
+    return [
+        DefaultInfo(
+            files = depset(ctx.files.srcs),
+        ),
+        RustStdLibInfo(
+            std_rlibs = std_rlibs,
+            dot_a_files = dot_a_files,
+            between_alloc_and_core_files = between_alloc_and_core_files,
+            core_files = core_files,
+            between_core_and_std_files = between_core_and_std_files,
+            std_files = std_files,
+            alloc_files = alloc_files,
+        ),
+    ]
+
+rust_stdlib_filegroup = rule(
+    doc = "A dedicated filegroup-like rule for Rust stdlib artifacts.",
+    implementation = _rust_stdlib_filegroup_impl,
+    attrs = {
+        "srcs": attr.label_list(
+            allow_files = True,
+            doc = "The list of targets/files that are components of the rust-stdlib file group",
+            mandatory = True,
+        ),
+    },
+)
+
+def _ltl(library, ctx, cc_toolchain, feature_configuration):
+    return cc_common.create_library_to_link(
+        actions = ctx.actions,
+        feature_configuration = feature_configuration,
+        cc_toolchain = cc_toolchain,
+        static_library = library,
+        pic_static_library = library,
+    )
+
+def _make_libstd_and_allocator_ccinfo(ctx, rust_lib, allocator_library):
+    """Make the CcInfo (if possible) for libstd and allocator libraries.
+
+    Args:
+        ctx (ctx): The rule's context object.
+        rust_lib: The rust standard library.
+        allocator_library: The target to use for providing allocator functions.
+
+
+    Returns:
+        A CcInfo object for the required libraries, or None if no such libraries are available.
+    """
+    cc_toolchain, feature_configuration = find_cc_toolchain(ctx)
+    link_inputs = []
+
+    if not RustStdLibInfo in ctx.attr.rust_lib:
+        fail(dedent("""\
+            {} --
+            The `rust_lib` must be a target providing `rust_common.stdlib_info` 
```

Comment:
I took the liberty of rephrasing the error message. Please take a look and feel free to change further :)